src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx

```diff
@@ -32,18 +44,52 @@ export const ActivityFlowBuilder = () => {
   );
   const activities: AppletFormValues['activities'] = watch('activities');
 
+  const handleFlowActivityDuplicate = (index: number) => {
+    if (!activityFlowItems) return;
+    insert(index + 1, { activityId: activityFlowItems[index].activityId, id: uuidv4() });
+  };
+
+  const handleFlowActivityToDeleteSet = (index: number, name: string) => () =>
+    setFlowActivityToDeleteData({ index, name });
+
+  const handleFlowActivityDelete = () => {
+    if (!flowActivityToDeleteData) return;
+
+    remove(flowActivityToDeleteData.index);
+    setFlowActivityToDeleteData(null);
+  };
+
+  const handleFlowActivityAdd = (activityKey: string) =>
+    append({ id: uuidv4(), activityId: activityKey });
+
+  const handleFlowActivityToUpdateSet = (event: MouseEvent<HTMLElement>, index: number) => {
+    setIndexToUpdate(index);
+    let parentElement = event.currentTarget.parentNode as HTMLElement;
+    while (parentElement && !parentElement.classList.contains('builder-item')) {
```

Comment:
could you move `builder-item` in a separate const file?

Reply:
done